### PR TITLE
fetch survey modules through GitHub  API (authenticated)

### PR DESCRIPTION
**Pull request by JoeArmani**
Related: Quest 2.0 improvements
https://github.com/episphere/connect/issues/1066

• Fetch survey data (questionnaire repo) from GitHub API with an authenticated request.
This should be more robust than using the raw file fetch and should eliminate response code 0 errors.

Note: Merge with ConnectApp PR: https://github.com/episphere/connectApp/pull/811

### CODE REVIEW

File: utils/connectApp.js [hunk 1]
```diff
@@ -74,6 +74,28 @@ const connectApp = async (req, res) => {
       return res.status(200).json({data: shaResult, code: 200});
     }
 
+    else if (api === 'getQuestSurveyFromGitHub') {
+      if (req.method !== 'GET') {
+        return res.status(405).json(getResponseJSON('Only GET requests are accepted!', 405));
+      }
+
+      if (!req.query.sha || req.query.sha === '') {
```
Comment (amber-emmes) [inline utils/connectApp.js:82]: This check is redundant; the empty string is already falsey.

**Review by amber-emmes — APPROVED** [review]

File: utils/connectApp.js (hunk 1 above)
Comment (JoeArmani) [reply utils/connectApp.js]: Good call. I cleaned it up. Thanks!

File: utils/submission.js [hunk 2]
```diff
@@ -582,17 +582,29 @@ const getUserCollections = async (req, res, uid) => {
 }
 
 /**
- * Get the sha of a module in the questionnaire repository.
- * @param {string} path - the path to the module in the questionnaire repository.
- * @returns {string} - the sha of the module.
+ * Fetch the GitHub API token from Secret Manager.
+ * @returns {String} - The GitHub API token.
```
Comment (we-ai) [inline utils/submission.js:586]: Type should be `string`. Please double check.

File: utils/connectApp.js [hunk 3]
```diff
@@ -74,12 +74,34 @@ const connectApp = async (req, res) => {
...
+
+      if (!req.query.sha) {
+        return res.status(400).json(getResponseJSON('Sha parameter is required!', 400));
+      }
+
+      if (!req.query.path) {
+        return res.status(400).json(getResponseJSON('Path parameter is required!', 400));
+      }
+
+      const sha = req.query.sha;
```
Comment (we-ai) [inline utils/connectApp.js:90]: Can we use `const {sha, path} = req.query;`?

File: utils/submission.js [hunk 4]
```diff
@@ -682,6 +691,24 @@ const getSHAFromGitHubCommitData = async (surveyStartTimestamp, path) => {
     }
 }
 
+/**
+ * Fetch a raw file from the questionnaire repository using the GitHub API.
+ * @param {String} sha - The SHA of the raw file commit to fetch.
+ * @param {String} path - The path to the file in the questionnaire repository.
+ * @param {String} token - The GitHub API token.} surveyStartTimestamp
```
Comment (we-ai) [inline utils/submission.js:698]: `string`

Comment (we-ai) [inline utils/submission.js:698]:
```suggestion
 * @param {String} token - The GitHub API token. surveyStartTimestamp 
```

**Review by we-ai — APPROVED** [review]
Looks good.
Commented on minor things.

File: utils/connectApp.js (hunk 3 above)
Comment (JoeArmani) [reply utils/connectApp.js:90]: sure, updated to use destructuring

**Pull request merged**